### PR TITLE
Allow image removal from reader studies

### LINKED ISSUE #1059: Deleting images from reader studies
I added a list of images to my reader study on grand-challenge, but it appears that I cannot delete them. Could this functionality be added to grand-challenge? Many thanks in advance.

### PR DESCRIPTION
**Pull request by MikeOverkamp-diag**
Closes #1059

### CODE REVIEW

File: app/grandchallenge/reader_studies/templates/reader_studies/readerstudy_detail.html [hunk 1]
```diff
@@ -190,7 +190,28 @@ <h2>Upload Sessions</h2>
...
+                        </div>
+                        <div class="col-3 mb-1">
+                            <div class="d-flex justify-content-end align-items-center">
+                                <button class="btn btn-danger remove-image" data-id="{{ image.id }}">Remove</button>
+                            </div>
+                        </div>
+                    </div>
+                </li>
+            {% endfor %}
+        </ul>
```
Comment (jmsmkn) [inline app/grandchallenge/reader_studies/templates/reader_studies/readerstudy_detail.html]: I think that this needs to be moved to a separate page, we have some studies with 1000's of images.

File: app/grandchallenge/reader_studies/views.py [hunk 2]
```diff
@@ -398,6 +398,27 @@ def generate_hanging_list(self, request, pk=None):
...
+        image_id = request.data.get("image")
+        reader_study = self.get_object()
+        try:
+            reader_study.images.remove(Image.objects.get(id=image_id))
+            messages.add_message(
+                request, messages.SUCCESS, "Image removed from reader study."
+            )
+            return Response({"status": "Image removed from reader study."},)
+        except Image.DoesNotExist:
+            pass
```
Comment (jmsmkn) [inline app/grandchallenge/reader_studies/views.py]: It's best not to do try-except-pass. Replace `pass` with the fall through block (L413-420) and return in the except block.

File: app/grandchallenge/reader_studies/views.py [hunk 3]
```diff
@@ -398,6 +398,27 @@ def generate_hanging_list(self, request, pk=None):
         )
         return Response({"status": "Hanging list generated."},)
 
+    @action(detail=True, methods=["patch"])
+    def remove_image(self, request, pk=None):
```
Comment (jmsmkn) [inline app/grandchallenge/reader_studies/views.py]: What are the permissions on this endpoint?

Comment (MikeOverkamp-diag) [reply app/grandchallenge/reader_studies/views.py]: You need edit rights. Added a test for this.

**Comment by MikeOverkamp-diag** [comment]
Rebased on master, hence the force push. No additional changes made.

**Comment by jmsmkn** [comment]
Could you rebase again? I'm done changing the templates now :-)

**Comment by MikeOverkamp-diag** [comment]
Rebased and added pagination in https://github.com/comic/grand-challenge.org/pull/1080/commits/aff741c56a3da58f4a246fe0b1157e3c9d37c5c2

**Pull request merged**